Prepare front-door for Tech Preview (#697)
unicode-org/message-format-wg: README.md

- [Goals and Non-Goals](docs/goals.md)
- [Record of Consensus Decisions](docs/consensus_decisions.md)

## MessageFormat 2 Technical Preview

The MessageFormat 2 specification was accepted by the CLDR-TC on 2024-02-28
as a new part of the LDML specification.
This specification is being released as a "Tech Preview", 
which means that the stability policy is not in effect and feedback from
users and implementers might result in changes to the syntax, data model,
functions, or other normative aspects of MessageFormat 2.
Such changes are expected to be minor and, to the extent possible,
to be compatible with what is defined in the Tech Preview.

The MFWG welcomes any and all feedback, including bugs reports, implementation
reports, success stories, feature requests, requests for clarification, 
or anything that would be helpful in stabilizing the specification and
promoting widespread adoption.

The MFWG specifically requests feedback on the following issues:

> - How to perform non-integer exact number selection (#675)

Reviewer: Give full urls here for portability

- Whether `markup` should support additional spaces (#650)
- Whether "attribute-like" behavior is needed and what form it should take (#642)
- Whether to relax constraints on complex message start (#610)
- Whether omitting the `*` variant key should be permitted (#603)

## What is MessageFormat 2?

> MessageFormat 2 defines the data model, syntax, processing, and conformance requirements 

Reviewer 2: I would start with a short lead-in that isn't so technical. Maybe something like

Software needs to construct messages that incorporate various pieces of information. The complexities of the world's languages make this challenging.

for the next generation of dynamic messages. 

> It is intended for adoption by programming languages and APIs. 

Reviewer 2: ```suggestion
It is intended for adoption by programming languages, software libraries, and software localization tooling.
```
I think it is worth calling this out specifically.

It enables the integration of internationalization APIs (such as date and number formats), 
grammatical matching (such as plurals or genders), 
as well as user-defined formats and message selectors,
so that developers and translators can create natural, grammatically-correct,

> fluent user interfaces.

Reviewer 2: I would stay away from 'fluent'; confusing to programmers who think of 'fluent APIs'

- [Goals and Non-Goals](docs/goals.md)
- [Record of Consensus Decisions](docs/consensus_decisions.md)

## MessageFormat 2 Technical Preview

> The MessageFormat 2 specification was accepted by the CLDR-TC on 2024-02-28
> as a new part of the LDML specification.

Reviewer 3: We care about the mechanics here, but the readers of this probably don't. Also, linkify.

```suggestion
The MessageFormat 2 specification is a new part of
the [LDML](https://www.unicode.org/reports/tr35/) specification.
```

> This specification is being released as a "Tech Preview", 

Reviewer 3: ```suggestion
This specification is initially released as a "Tech Preview", 
```

- Whether `markup` should support additional spaces [#650](https://github.com/unicode-org/message-format-wg/issues/650)
- Whether "attribute-like" behavior is needed and what form it should take [#642](https://github.com/unicode-org/message-format-wg/issues/642)
- Whether to relax constraints on complex message start [#610](https://github.com/unicode-org/message-format-wg/issues/610)
- Whether omitting the `*` variant key should be permitted [#603](https://github.com/unicode-org/message-format-wg/issues/603)

## What is MessageFormat 2?

Software needs to construct messages that incorporate various pieces of information. 
The complexities of the world's languages make this challenging.
MessageFormat 2 defines the data model, syntax, processing, and conformance requirements 
for the next generation of dynamic messages. 
It is intended for adoption by programming languages, software libraries, and software localization tooling.
It enables the integration of internationalization APIs (such as date or number formats),
and grammatical matching (such as plurals or genders). 

> It is extensible, allowing software developers to create formatting
> or message selection logic that add on to the core capabilities.

Reviewer 3: ```suggestion
It is extensible, allowing software developers to create formatting
or message selection logic that add on to the core capabilities.
Its data model is — as far as we can tell — capable of
representing messages in all existing syntaxes,
enabling gradual adoption by users of other syntaxes.
```

Author: I'm reluctant to commit this as written because it doesn't match in tone. Perhaps:

```suggestion
It is extensible, allowing software developers to create formatting
or message selection logic that add on to the core capabilities.
Its data model provides a means of representing existing syntaxes,
thus enabling gradual adoption by users of older formatting systems.
```

> The goal is to allow developers and translators to create natural-sounding, grammatically-correct,

Reviewer 3: Could add a paragraph break here?

```suggestion

The goal is to allow developers and translators to create natural-sounding, grammatically-correct,
```

which means that the stability policy is not in effect and feedback from
users and implementers might result in changes to the syntax, data model,
functions, or other normative aspects of MessageFormat 2.
Such changes are expected to be minor and, to the extent possible,
to be compatible with what is defined in the Tech Preview.

The MFWG welcomes any and all feedback, including bugs reports, implementation
reports, success stories, feature requests, requests for clarification, 
or anything that would be helpful in stabilizing the specification and
promoting widespread adoption.

The MFWG specifically requests feedback on the following issues:
- How to perform non-integer exact number selection [#675](https://github.com/unicode-org/message-format-wg/issues/675)
- Whether `markup` should support additional spaces [#650](https://github.com/unicode-org/message-format-wg/issues/650)
- Whether "attribute-like" behavior is needed and what form it should take [#642](https://github.com/unicode-org/message-format-wg/issues/642)
- Whether to relax constraints on complex message start [#610](https://github.com/unicode-org/message-format-wg/issues/610)

> - Whether omitting the `*` variant key should be permitted [#603](https://github.com/unicode-org/message-format-wg/issues/603)

Author: Thanks for this. The list here consists strictly of items tagged with "seek feedback" label. I agree with most of your observations and will add some details shortly.

> I personally would also like feedback on the desirability of syntax divergence between simple and complex messages (which is more general than https://github.com/unicode-org/message-format-wg/issues/610 in its current form, cf. https://github.com/unicode-org/message-format-wg/issues/610#issuecomment-1902686867 and its outbound link tree). I guess the latter might be https://github.com/unicode-org/message-format-wg/issues/512?

I think these are details of the syntax? Wouldn't feedback about the syntax in general work better? How intuitive (or unintuitive) it is to use, whether simple-vs-complex is easy to grok, etc.

> - Whether omitting the `*` variant key should be permitted [#603](https://github.com/unicode-org/message-format-wg/issues/603)

Reviewer 4: > I think these are details of the syntax? Wouldn't feedback about the syntax in general work better? How intuitive (or unintuitive) it is to use, whether simple-vs-complex is easy to grok, etc.

Yes, but we should have specific prompts to get specific feedback.

> - Whether omitting the `*` variant key should be permitted [#603](https://github.com/unicode-org/message-format-wg/issues/603)

Author: Always happy to get suggestions 😸
